app/models/manageiq/providers/amazon/inventory/parser/network_manager.rb

```diff
@@ -16,6 +19,38 @@ def parse
 
   private
 
+  def build_and_index_network_routers
+    # We need to pre-calculate NetworkRouter relations, because AWS Subnet api doesn't offer us a foreign_key for each
+    # router
+    @indexed_routers = {:cloud_subnets_ems_ref => {}, :cloud_networks_ems_ref => {}}
+
+    collector.network_routers.each do |network_router|
+      uid              = network_router['route_table_id']
+      main_route_table = false
+      network_router['associations'].each do |association|
+        network_router_lazy = persister.network_routers.lazy_find(uid)
+        if association['main']
+          main_route_table                                                    = true
+          @indexed_routers[:cloud_networks_ems_ref][network_router['vpc_id']] = network_router_lazy
+        else
+          @indexed_routers[:cloud_subnets_ems_ref][association['subnet_id']] = network_router_lazy
+        end
+      end
+
+      persister.network_routers.find_or_build(uid).assign_attributes(
+        :status           => network_router['routes'].all? { |x| x['state'] == 'active' } ? 'active' : 'inactive',
+        :name             => get_from_tags(network_router, "Name") || uid,
+        :extra_attributes => {
+          :main_route_table => main_route_table,
+          :routes           => network_router['routes'],
+          :propagating_vgws => network_router['propagating_vgws']
+        },
+        # TODO(lsmola) model orchestration_stack_id relation for a NetworkRouter
+        # :orchestration_stack => persister.orchestration_stacks.lazy_find(get_from_tags(network_router, "aws:cloudformation:stack-id")),
```

Comment:
hehe, we need to add model change, so we can see routers created as part of a stack. Can we take it as a part of comment rather than inactive code? Makes the todo clearer for me. :-)

the extra attributes are then mapped to virtual columns in the model

Reply:
sure